packages/core/admin/admin/src/core/apis/rbac.ts

```diff
@@ -0,0 +1,58 @@
+import { Location } from 'react-router-dom';
+
+import type { Permission, User } from '../../features/Auth';
+
+interface RBACContext extends Pick<Location, 'pathname' | 'search'> {
+  /**
+   * The current user.
+   */
+  user?: User;
+  /**
+   * The permissions of the current user.
+   */
+  permissions: Permission[];
+}
+
+interface RBACMiddleware {
+  (
+    ctx: RBACContext
+  ): (
+    next: (permissions: Permission[]) => Promise<Permission[]> | Permission[],
```

Comment:
doesn't that indicate that the context is mutable? Do you foresee an issue with that? 🤔

Reply:
Re: other comment thread, it looks like redux middlware are completely curried so it would be like context => next => premissions => {}, would that make sense here too?